### ISSUE TITLE
Switch to Pytorch's automatic mixed precision

### ISSUE THREAD

**Issue by s-jse** [stanford-oval/genienlp#52]
Pytorch 1.7 added support for mixed precision training and evaluation ([https://pytorch.org/docs/stable/amp.html?highlight=amp#module-torch.cuda.amp](https://pytorch.org/docs/stable/amp.html?highlight=amp#module-torch.cuda.amp)).

This will be especially useful for speeding up inference for both semantic parsing and paraphrase generation. Also, training with mixed precision -if experiments show that it does not degrade accuracy- can help speed up our pipeline.
